### PR TITLE
Make ClerkProvider idempotent during SSR and CSR

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

### CODE REVIEW

File: packages/react/src/contexts/ClerkProvider.tsx [hunk 1]
```diff
@@ -15,24 +14,15 @@ export interface ClerkProviderProps extends IsomorphicClerkOptions {
...
-    void clerk.loadClerkJS().then(() => setClerkLoaded(true));
-  }, []);
+  const { initialState, children, Clerk, frontendApi, ...options } = props;
 
   return (
     <StructureContext.Provider value={StructureContextStates.noGuarantees}>
       <ClerkContextProvider
-        isomorphicClerk={clerk}
-        clerkLoaded={clerkLoaded}
+        initialState={initialState}
```
Comment (igneel64) [inline packages/react/src/contexts/ClerkProvider.tsx:22]: Thats the fix 🎸

Comment (nikosdouvlis) [reply packages/react/src/contexts/ClerkProvider.tsx:22]: Technically https://github.com/clerkinc/javascript/pull/225/commits/8f9481cf088c63b3cd3192cb1396596a98b11980 is needed as well!

File: packages/react/src/isomorphicClerk.ts [hunk 2]
```diff
@@ -59,36 +57,33 @@ export default class IsomorphicClerk {
...
   }
 
   static #instance: IsomorphicClerk;
 
   static getOrCreateInstance(params: NewIsomorphicClerkParams) {
-    if (!this.#instance) {
+    // During SSR: a new instance should be created for every request
+    // During CSR: use the cached instance for the whole lifetime of the app
+    // This method should be idempotent in both scenarios
+    if (!inClientSide() || !this.#instance) {
```
Comment (SokratisVidros) [inline packages/react/src/isomorphicClerk.ts:70]: 💯

File: packages/react/src/isomorphicClerk.ts [hunk 3]
```diff
@@ -59,36 +57,33 @@ export default class IsomorphicClerk {
...
     this.options = options;
     this.Clerk = Clerk;
-    this.initialState = initialState;
     this.mode = inClientSide() ? 'browser' : 'server';
+    void this.loadClerkJS();
   }
 
   async loadClerkJS(): Promise<BrowserClerk | undefined> {
-    if (this.#loaded) {
+    if (this.mode !== 'browser' || this.#loaded) {
```
Comment (SokratisVidros) [inline packages/react/src/isomorphicClerk.ts:86]: 🙃 You can use `isClientSide`

**Review by SokratisVidros — APPROVED** [review]
💯

**Pull request merged**